### PR TITLE
The removal of Orchard.FileSystem

### PR DESCRIPTION
**Pull request by Jetski5822**
This was built through my lack of understanding of IFIleInfo and IFileProvider.

I have changes some things.

1. Removed Orchard.FileSystem
2. Made use of IHostingEnvironment
3. Added app_data to ShellOptions... if you need app_data access, you can go through there.

### CODE REVIEW

File: src/Orchard.Environment.Extensions/ExtensionLibraryService.cs [hunk 1]
```diff
@@ -37,20 +37,19 @@ public class ExtensionLibraryService : IExtensionLibraryService
...
-            IAppDataFolder appDataFolder,
+            IOptions<ShellOptions> shellOptionsAccessor,
             ILogger<ExtensionLibraryService> logger)
         {
             _metadataReferences = new Lazy<List<MetadataReference>>(GetMetadataReferences);
             _applicationPartManager = applicationPartManager;
-            _fileSystem = fileSystem;
-            _probingFolderPath = appDataFolder.MapPath(ProbingDirectoryName);
+            _shellOptions = shellOptionsAccessor.Value;
+            _probingFolderPath = Path.Combine(_shellOptions.ShellHostContainer.PhysicalPath, ProbingDirectoryName);
```
Comment (sebastienros) [inline src/Orchard.Environment.Extensions/ExtensionLibraryService.cs]: what's the difference between `_shellOptions.ShellHostContainer.PhysicalPath` and `IHostEnvironment.ContentPath` ?

**Comment by Jetski5822** [comment]
Good point, I'm going to remove the ShellOptions one

File: src/Orchard.Environment.Extensions/ExtensionLibraryService.cs (hunk 1 above)
Comment (Jetski5822) [inline src/Orchard.Environment.Extensions/ExtensionLibraryService.cs]: done

**Comment by sebastienros** [comment]
Are you also removing IHostEnvrionment? Please do if not.

**Comment by Jetski5822** [comment]
Merged it in.

File: src/Orchard.Environment.Extensions/ExtensionLibraryService.cs [hunk 2]
```diff
@@ -37,20 +39,20 @@ public class ExtensionLibraryService : IExtensionLibraryService
...
+        private readonly IHostingEnvironment _hostingEnvironment;
         private readonly string _probingFolderPath;
         private readonly ILogger _logger;
 
         public ExtensionLibraryService(
             ApplicationPartManager applicationPartManager,
-            IOrchardFileSystem fileSystem,
-            IAppDataFolder appDataFolder,
+            IHostingEnvironment hostingEnvironment,
+            IOptions<ShellOptions> optionsAccessor,
```
Comment (sebastienros) [inline src/Orchard.Environment.Extensions/ExtensionLibraryService.cs]: Why is it here?

File: src/Orchard.Environment.Shell/ShellOptions.cs [hunk 3]
```diff
@@ -1,7 +0,0 @@
-﻿namespace Orchard.Environment.Shell
-{
-    public class ShellOptions
```
Comment (sebastienros) [inline src/Orchard.Environment.Shell/ShellOptions.cs]: Not sure it should have changed. This is becoming the options for when the shells are loaded from the file system in "Sites", it's an option for a specific implementation.

**Pull request merged**